### PR TITLE
progress

### PR DESCRIPTION
**Pull request by ashishrudra**
progress

### CODE REVIEW

File: app/models/requirement.rb [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
+  include AASM
   belongs_to :deliverable
   validates :name, { presence: true }
   validates :deliverable, { presence: true }
 
   before_validation :set_defaults
 
+  enum status: [:created, :started, :completed]
+
+  aasm column: :status do
```
Comment (stevenweber) [inline app/models/requirement.rb]: Do we need a several states for this -- what about inferring 'started' based on whether someone has picked it up and just having a 'completed' boolean?

File: app/views/deliverables/_form.html.haml [hunk 2]
```diff
@@ -16,4 +16,4 @@
     = f.text_field(:value)
 
   .actions
-    = f.submit 'Add'
+    = f.submit 'Save'
```
Comment (stevenweber) [inline app/views/deliverables/_form.html.haml]: `t("actions.save")`

File: app/views/layouts/application.html.haml [hunk 3]
```diff
@@ -6,8 +6,11 @@
     %link{ href: "http://fonts.googleapis.com/css?family=Oswald:400,700", rel: "stylesheet", type: "text/css" }
     %link{ href: "http://fonts.googleapis.com/css?family=Open+Sans+Condensed:300", rel: "stylesheet", type: "text/css" }
     %link{ href: "http://fonts.googleapis.com/css?family=Special+Elite", rel: "stylesheet", type: "text/css" }
+    %link{ href: "http://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css", :rel => "stylesheet"}
```
Comment (stevenweber) [inline app/views/layouts/application.html.haml]: Do we need bootstrap?

File: app/views/requirements/_form.html.haml [hunk 4]
```diff
@@ -15,5 +15,9 @@
     = f.label(:description)
     = f.text_field(:description)
 
+  .fill-me-in
+    = f.label("Estimate in days")
```
Comment (stevenweber) [inline app/views/requirements/_form.html.haml]: Use can use [active record translations](http://stackoverflow.com/questions/10812774/how-can-i-translate-the-activerecord-attribute-name-in-rails-3-2) here.

File: app/views/deliverables/show.html.haml [hunk 5]
```diff
@@ -1,6 +1,15 @@
 - if notice
   %p#notice= notice
 
+- if !@deliverable.estimate.zero?
+  .progress
```
Comment (stevenweber) [inline app/views/deliverables/show.html.haml]: I really like this.

File: app/views/deliverables/show.html.haml [hunk 6]
```diff
@@ -1,6 +1,15 @@
...
+    .progress-bar.progress-bar-success.progress-bar-striped.active{:role => "progressbar", :style => "width:#{@deliverable.progress["completed"]}%"}
+      Completed - #{@deliverable.progress["completed"]}%
```
Comment (stevenweber) [inline app/views/deliverables/show.html.haml]: `t`

File: app/assets/stylesheets/forms.css [hunk 7]
```diff
@@ -33,6 +33,19 @@ input {
   color: rgba(255, 255, 255, 0.6);
 }
 
+.fill-me-in input[type=number] {
```
Comment (stevenweber) [inline app/assets/stylesheets/forms.css]: Were there problems just adding this to the text one?

```css
.fill-me-in input[type=text],
.fill-me-in input[type=number] 
```

File: app/views/deliverables/show.html.haml [hunk 8]
```diff
@@ -1,6 +1,15 @@
 - if notice
   %p#notice= notice
 
+- if !@deliverable.estimate.zero?
+  .progress
+    .progress-bar.progress-bar-success.progress-bar-striped.active{:role => "progressbar", :style => "width:#{@deliverable.progress["completed"]}%"}
```
Comment (stevenweber) [inline app/views/deliverables/show.html.haml]: What do you think about this for colors?

```
Unstarted: very-transparent white
Started: the medium-transparent white
Completed: same green
```

**Comment by stevenweber** [comment]
This is very exciting, great idea.

File: app/models/requirement.rb (hunk 1 above)
Comment (ashishrudra) [inline app/models/requirement.rb]: first i started with boolean. then changed it to aasm thinking when someone picks a task/requirement, it should keep that record by calling ".start!" and in the callback , we can implement some notification logic for other members on the mission.

File: app/views/layouts/application.html.haml (hunk 3 above)
Comment (ashishrudra) [inline app/views/layouts/application.html.haml]: its for the progress bar.. i will extract that css later

File: app/views/requirements/_form.html.haml (hunk 4 above)
Comment (ashishrudra) [inline app/views/requirements/_form.html.haml]: :+1:  done

**Pull request merged**